src/core/simd.d

```diff
@@ -731,17 +779,48 @@ version (D_SIMD)
                 }
             }
 
-            test!void16();
-            test!byte16();
-            test!ubyte16();
-            test!short8();
-            test!ushort8();
-            test!int4();
-            test!uint4();
-            test!long2();
-            test!ulong2();
-            test!double2();
-            test!float4();
+            static string size1Tests(string T)()
+            {
+                string res;
+                res ~= "test!(__vector("~T~"[1]))();";
+                res ~= "test!(__vector("~T~"[2]))();";
+                res ~= "test!(__vector("~T~"[4]))();";
+                return res;
+            }
+
+            static string generalTests(string T, string size)()
+            {
+                string res;
+                res ~= "test!(__vector("~T~"[8 / "~size~"]))();";
```

Comment:
IMO, we can do without these extra tests (the `loadUnaligned` tests haven't been extended). Sorry if I put you on the wrong track; I said you'd need to generalize the tests *if* you want to test non-128-bit vectors too. If you do want to keep the tests, then the mixin orgy and passing type and size as strings can be handled more elegantly:
```D
static void generalTests(T)()
{
    static foreach (size; [8, 16, 32, 64])
        test!(__vector(T[size / T.sizeof]));
}

generalTests!void();
generalTests!byte();
...
```

Reply:
Cool indeed, I haven't thought about it. TBH, since we moved the code to `ldc/simd`, I don't know if there is any point in adding anything to `core/simd`.